apps/src/code-studio/hiddenStageRedux.js

```diff
@@ -101,23 +134,56 @@ export function getHiddenStages(scriptName, canHideStages) {
       url: `/s/${scriptName}/hidden_stages`,
       dataType: 'json',
       contentType: 'application/json'
-    }).done(response => {
-      // For a teacher, we get back a map of section id to hidden stage ids
-      // For a student, we just get back a list of hidden stage ids. Turn that
-      // into an object, under the 'sectionId' of STUDENT_SECTION_ID
-      if (Array.isArray(response)) {
-        response = { [STUDENT_SECTION_ID]: response };
-      }
-
-      Object.keys(response).forEach(sectionId => {
-        const hiddenStageIds = response[sectionId];
-        hiddenStageIds.forEach(stageId => {
-          dispatch(updateHiddenStage(sectionId, stageId, true));
-        });
+    })
+    .done(response => dispatch(initializeHiddenStages(response, canHideStages)))
+    .fail(err => console.error(err));
+  };
+}
+
+/**
+ * Initialize hidden stages based on server data. In the case of a student, this
+ * will be a list list of hidden stage ids. In the case of a teacher, it will be
+ * a mapping from section id to a list of hidden stage ids for that section
+ * @param {string[]|Object<string, string[]} data
+ * @param {boolean} canHideStages - True if we're able to toggle hidden stages
+ */
+function initializeHiddenStages(data, canHideStages) {
+  return dispatch => {
+    // For a teacher, we get back a map of section id to hidden stage ids
+    // For a student, we just get back a list of hidden stage ids. Turn that
+    // into an object, under the 'sectionId' of STUDENT_SECTION_ID
+    if (Array.isArray(data)) {
+      data = { [STUDENT_SECTION_ID]: data };
+    }
+
+    Object.keys(data).forEach(sectionId => {
+      const hiddenStageIds = data[sectionId];
+      hiddenStageIds.forEach(stageId => {
+        dispatch(updateHiddenStage(sectionId, stageId, true));
+      });
+    });
+    dispatch(setHiddenStagesInitialized(!!canHideStages));
+  };
+}
+
+/**
+ * @param {function} dispatch - Method for dispatching actions to redux
+ * @param {string[]|Object<string, string[]} data
```

Comment:
This doc is wrong; `dispatch` isn't a param to `initializeHiddenScripts`, it's a param to the returned function.  Could doc this way, I think?
```js
/**
 * @param {string[]|Object<string, string[]>} data
 * @return {function(dispatch:function)} a thunk action
 */
```
Also missing a `>` in the type here too.

Reply:
Nice catch. I was originally passing in dispatch, and then realized that was silly since we already have a pattern around returning curried functions.